sympy/integrals/transforms.py

```diff
@@ -1186,606 +1194,540 @@ def _laplace_build_rules(t, s):
-    (t**(b)*besselj(n, 2*sqrt(a*t)),
-     a**(n/2)*s**(-n-1)*exp(-a/s),
-     And(And(a>0, n>-1), Eq(b, n*S.Half)), S.Zero, dco),
-    # 8.9
-    (besselj(0, a*sqrt(t**2+b*t)),
-     exp(b*s-b*sqrt(s**2+a**2))/sqrt(s**2+a**2),
-     b>0, S.Zero, dco),
-    # 8.10, 8.11
-    (besseli(n, a*t),
-     a**n/(sqrt(s**2-a**2)*(s+sqrt(s**2-a**2))**n),
-     And(a>0, n>-1), Abs(a), dco),
-    # 8.12
+     And(re(n)>-1, Eq(b, n+1)), Abs(im(a)), dco), # 4.14.8
+    (besselj(0, 2*sqrt(a*t)), exp(-a/s)/s,
+     S.true, S.Zero, dco), # 4.14.25
+    (t**(b)*besselj(n, 2*sqrt(a*t)), a**(n/2)*s**(-n-1)*exp(-a/s),
+     And(re(n)>-1, Eq(b, n*S.Half)), S.Zero, dco), # 4.14.30
+    (besselj(0, a*sqrt(t**2+b*t)), exp(b*s-b*sqrt(s**2+a**2))/sqrt(s**2+a**2),
+     Abs(arg(b))<pi, Abs(im(a)), dco), # 4.15.19
+    (besseli(n, a*t), a**n/(sqrt(s**2-a**2)*(s+sqrt(s**2-a**2))**n),
+     re(n)>-1, Abs(re(a)), dco), # 4.16.1
     (t**b*besseli(n, a*t),
      2**n/sqrt(pi)*gamma(n+S.Half)*a**n*(s**2-a**2)**(-n-S.Half),
-     And(And(a>0, n>-S.Half), Eq(b, n)), Abs(a), dco),
-    # 8.13
+     And(re(n)>-S.Half, Eq(b, n)), Abs(re(a)), dco), # 4.16.6
     (t**b*besseli(n, a*t),
      2**(n+1)/sqrt(pi)*gamma(n+S(3)/2)*a**n*s*(s**2-a**2)**(-n-S(3)/2),
-     And(And(a>0, n>-1), Eq(b, n+1)), Abs(a), dco),
-    # 8.15, 8.16
-    (t**(b)*besseli(n, 2*sqrt(a*t)),
-     a**(n/2)*s**(-n-1)*exp(a/s),
-     And(And(a>0, n>-1), Eq(b, n*S.Half)), S.Zero, dco),
-    # 8.17
-    (bessely(0, a*t),
-     -2/pi*asinh(s/a)/sqrt(s**2+a**2),
-     a>0, S.Zero, dco),
-    # 8.18
-    (besselk(0, a*t),
-     (log(s+sqrt(s**2-a**2)))/(sqrt(s**2-a**2)),
-     a>0, Abs(a), dco)
+     And(re(n)>-1, Eq(b, n+1)), Abs(re(a)), dco), # 4.16.7
+    (t**(b)*besseli(n, 2*sqrt(a*t)), a**(n/2)*s**(-n-1)*exp(a/s),
+     And(re(n)>-1, Eq(b, n*S.Half)), S.Zero, dco), # 4.16.18
+    (bessely(0, a*t), -2/pi*asinh(s/a)/sqrt(s**2+a**2),
+     S.true, Abs(im(a)), dco), # 4.15.44
+    (besselk(0, a*t), log((s + sqrt(s**2-a**2))/a)/(sqrt(s**2-a**2)),
+     S.true, -re(a), dco) # 4.16.23
     ]
     return laplace_transform_rules
 
-def _laplace_cr(f, a, c, **hints):
+def _laplace_cr(f, **hints):
     """
-    Internal helper function that will return `(f, a, c)` unless `**hints`
-    contains `noconds=True`, in which case it will only return `f`.
+    Internal helper function that will return ``(f, a, c)`` unless ``**hints``
+    contains ``noconds=True``, in which case it will only return ``f``.
     """
-    conds = not hints.get('noconds', False)
-    if conds:
-        return f, a, c
+    noconds = hints.get('noconds', False)
+    if noconds and type(f) is tuple:
+        return f[0]
     else:
         return f
 
-def _laplace_rule_timescale(f, t, s, doit=True, **hints):
-    r"""
-    This internal helper function tries to apply the time-scaling rule of the
-    Laplace transform and returns `None` if it cannot do it.
+def _laplace_ct(f):
+    """
+    Internal helper function that completes the tuple as follows:
```

Comment:
Given that `_compute_transform` already returns a tuple now is this function needed?

Reply:
Yes, I use it in many places to create standard-valued tuples.

Reply:
The premise of this function is that it tries to hide inconsistent use of types in the code here. It would be better to go to the source and make sure that tuples are already being used everywhere.

Reply:
They were used everywhere, so it was easy to remove `_laplace_ct`. Also change (P)